pkg/cmd/release-payload-controller/batchjob_controller.go

```diff
@@ -0,0 +1,279 @@
+	batchJob, ok := obj.(*batchv1.Job)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("unable to cast batchJob: %v", obj))
+		return
+	}
+
+	klog.V(4).Infof("Processing batchJob: '%s/%s'", batchJob.Namespace, batchJob.Name)
+
+	tag, ok := batchJob.Annotations[releasecontroller.ReleaseAnnotationReleaseTag]
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("unable to process batchJob, missing '%s' annotation", releasecontroller.ReleaseAnnotationReleaseTag))
+		return
+	}
+
+	releasePayloadKey := fmt.Sprintf("%s/%s", c.releasePayloadNamespace, tag)
+	klog.V(4).Infof("Queueing ReleasePayload: %s", releasePayloadKey)
+	c.queue.Add(releasePayloadKey)
+}
+
+func (c *BatchJobStatusController) Run(ctx context.Context) {
+	defer utilruntime.HandleCrash()
+
+	klog.Info("Starting BatchJob Status Controller")
+	defer func() {
+		klog.Info("Shutting down BatchJob Status Controller")
+		c.queue.ShutDown()
+		klog.Info("BatchJob Status Controller shut down")
+	}()
+
+	if !cache.WaitForNamedCacheSync("BatchJobStatusController", ctx.Done(), c.cachesToSync...) {
+		return
+	}
+
+	go func() {
+		wait.UntilWithContext(ctx, c.runWorker, time.Second)
+	}()
+
+	<-ctx.Done()
+}
+
+func (c *BatchJobStatusController) runWorker(ctx context.Context) {
+	for c.processNextItem(ctx) {
+	}
+}
+
+func (c *BatchJobStatusController) processNextItem(ctx context.Context) bool {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(key)
+
+	err := c.sync(ctx, key.(string))
+
+	if err == nil {
+		c.queue.Forget(key)
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("%v failed with : %w", key, err))
+	c.queue.AddRateLimited(key)
+
+	return true
+}
+
+func (c *BatchJobStatusController) sync(ctx context.Context, key string) error {
+	klog.V(4).Infof("Starting BatchJobStatusController sync")
+	defer klog.V(4).Infof("BatchJobStatusController sync done")
+
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	klog.V(4).Infof("Processing ReleasePayload: '%s/%s' from workQueue", namespace, name)
+
+	// Get the ReleasePayload resource with this namespace/name
+	originalReleasePayload, err := c.releasePayloadLister.ReleasePayloads(namespace).Get(name)
+	// The ReleasePayload resource may no longer exist, in which case we stop processing.
+	if errors.IsNotFound(err) {
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	// If the release creation job status is terminal, then we have noting else to do
+	if len(originalReleasePayload.Status.ReleaseCreationJobResult.Status) > 0 && originalReleasePayload.Status.ReleaseCreationJobResult.Status != v1alpha1.ReleaseCreationJobUnknown {
+		return nil
+	}
+
+	// Lookup the job
+	job, err := c.batchJobLister.Jobs(c.batchJobNamespace).Get(originalReleasePayload.Name)
+	if errors.IsNotFound(err) {
+		return nil
```

Comment:
`// TODO consider a timeout`

Reply:
Added a `TODO`.  Is this something that can be done with a `Lister`?  I looked around and couldn't find anything that looked like it would handle a timeout.